### DISCUSSION TITLE
[Idea] Make invoice address optional

### DISCUSSION THREAD

**Discussion by rmilecki** [PrestaShop/PrestaShop#31068] [Ideas]
### The problem

In many Polish shops a majority of orders get documented by **receipts**. Those are kind of anonymous (no buyer info) documents and are printed using certificated & registered fiscal printers. For such purchases there is no need to get invoice address from a customer.

Can you make order **invoice** address **optional** in core PrestaShop, please?
That would allow checkout modules to ask customers if they actually need an invoice.

### The solution

Make `id_address_invoice` column in `orders` optional (allow `NULL`). Update relevant code to avoid PHP errors.

### Alternatives

It's rather hard to provide support for optional invoice address using a custom module. That would require setting `id_address_invoice` to some fake address and hacking order view page to hide it.

At the same time making `id_address_invoice` optional in core code sounds simple and shouldn't affect code maintenance noticeably.

### Additional context

In Poland there are two main kinds of documents that can be issued on purchase: **invoices** and **receipts**.

Invoices require specifying buyer name and address. Every issued invoice must be additionally software processed (reported to the government using a designed JPK summarize file). For small companies it means passing more documents to their accountants and paying extra money.

In most Polish shops most cutomsers don't need an invoice. You simply don't need a personal document (with your name on it) documenting your groceries shopping. Just like in most of real (non-Internet) shops in most of the world.

For above reasons most Polish shops simply print a fiscal document using a designed printer. It's low cost and gets automatically processed for tax purposes. Invoices are printed only per request for customers who actually need them.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by ghost** [comment]
good luck getting prestashop making this. I'm also with a problem, in Netherlands your shop must show shop information like Siret (france) in dutch KvK number. and more info you must have on site. but you have to write it your self. It's open source. And maybe they will look at it. 
if you post it here a could look at it if you mis something. i use the installation clean and just get writting. changing installing and  testing. maybe it helps you to.

kind regards Imke

**Comment by rmilecki** [comment]
My attempt for the first step: https://github.com/PrestaShop/PrestaShop/pull/31079

**Comment by ghost** [comment]
How did it go? Did they shoot you down? or did your idea make into the core?

**Reply by rmilecki** [reply]
It's an public pull request, you can see for yourself. Please follow the link I posted :)

**Comment by kpodemski** [comment]
Hi @rmilecki 

IMHO, it requires work on the back-end side and front-end sides. As discussed, I don't think we need any changes in the database structure. Would you like to meet to discuss available options? I'd be happy to help you. Later we could write a summary here on the topic :) Feel free to contact me on [Slack](https://www.prestashop-project.org/slack/) (@kpodemski).
